cmd/flags/cassandra/options.go

```diff
@@ -43,38 +56,104 @@ type Options struct {
 }
 
 // NewOptions creates a new Options struct.
-func NewOptions() *Options {
-	return &Options{
+func NewOptions(primaryNamespace string, otherNamespaces ...string) *Options {
+	options := &Options{
 		primary: &namespaceConfig{
 			Configuration: config.Configuration{
 				MaxRetryAttempts:   3,
 				Keyspace:           "jaeger_v1_local",
 				ProtoVersion:       4,
 				ConnectionsPerHost: 2,
 			},
-			servers: "127.0.0.1",
+			servers:   "127.0.0.1",
+			namespace: primaryNamespace,
 		},
-		others: make(map[string]*namespaceConfig),
+		others: make(map[string]*namespaceConfig, len(otherNamespaces)),
 	}
-}
 
-// Bind defines flags in the FlagSet for the primary namespace and optional other namespaces.
-func (opt *Options) Bind(flags *flag.FlagSet, primaryNamespace string, otherNamespaces ...string) {
-	bind(flags, primaryNamespace, opt.primary, &opt.primary.Configuration)
 	for _, namespace := range otherNamespaces {
-		nsCfg, ok := opt.others[namespace]
-		if !ok {
-			nsCfg = &namespaceConfig{}
-			opt.others[namespace] = nsCfg
-		}
-		bind(flags, namespace, nsCfg, nil)
+		options.others[namespace] = &namespaceConfig{namespace: namespace}
+	}
+
+	return options
+}
+
+// CreateFlags defines a set of flags prefixed with the namespace and binds them to cassandra
+// config struct, and a separate servers string pointer (because config struct uses array).
+func (opt *Options) CreateFlags(flagSet *flag.FlagSet) {
+	addFlags(flagSet, opt.primary)
+	for _, cfg := range opt.others {
+		addFlags(flagSet, cfg)
 	}
 }
 
+func addFlags(flagSet *flag.FlagSet, nsConfig *namespaceConfig) {
+	flagSet.Int(
+		nsConfig.namespace+suffixConnPerHost,
+		nsConfig.ConnectionsPerHost,
+		"The number of Cassandra connections from a single backend instance")
+	flagSet.Int(
+		nsConfig.namespace+suffixMaxRetryAttempts,
+		nsConfig.MaxRetryAttempts,
+		"The number of attempts when reading from Cassandra")
+	flagSet.Duration(
+		nsConfig.namespace+suffixTimeout,
+		nsConfig.Timeout,
+		"Timeout used for queries")
+	flagSet.String(
+		nsConfig.namespace+suffixServers,
+		nsConfig.servers,
+		"The comma-separated list of Cassandra servers")
+	flagSet.Int(
+		nsConfig.namespace+suffixPort,
+		nsConfig.Port,
+		"The port for cassandra")
+	flagSet.String(
+		nsConfig.namespace+suffixKeyspace,
+		nsConfig.Keyspace,
+		"The Cassandra keyspace for Jaeger data")
+	flagSet.Int(
+		nsConfig.namespace+suffixProtoVer,
+		nsConfig.ProtoVersion,
+		"The Cassandra protocol version")
+	flagSet.Duration(
+		nsConfig.namespace+suffixSocketKeepAlive,
+		nsConfig.SocketKeepAlive,
+		"Cassandra's keepalive period to use, enabled if > 0")
+}
+
+// InitFromViper parses the configuration properties
+func (opt *Options) InitFromViper(v *viper.Viper) {
+	// make sure that viper is initialized with default values for a case when CreateFlags is not
+	// passed to viper
+	a := &flag.FlagSet{}
+	opt.CreateFlags(a)
+	a.VisitAll(func(flag *flag.Flag) {
+		v.SetDefault(flag.Name, flag.DefValue)
+	})
```

Comment:
again, doesn't look right to have these here. And we should never need to call SetDefault.